packages/atomic/cypress/integration/facets/category-facet/category-facet-selectors.ts

```diff
@@ -8,49 +8,84 @@ export const BreadcrumbSelectors = {
 
 export const categoryFacetComponent = 'atomic-category-facet';
 export const CategoryFacetSelectors = {
-  shadow: () => cy.get(categoryFacetComponent).shadow(),
-  wrapper: () => CategoryFacetSelectors.shadow().find('[part="facet"]'),
-  labelButton: () =>
-    CategoryFacetSelectors.shadow().find('[part="label-button"]'),
-  placeholder: () =>
-    CategoryFacetSelectors.shadow().find('[part="placeholder"]'),
-  clearButton: () =>
-    CategoryFacetSelectors.shadow().find('[part="clear-button"]'),
-  allCategoriesButton: () =>
-    CategoryFacetSelectors.shadow().find('[part="all-categories-button"]'),
-  valueLabel: () =>
-    CategoryFacetSelectors.shadow().find('[part="value-label"]'),
-  values: () => CategoryFacetSelectors.shadow().find('[part="values"]'),
-  childValue: () => CategoryFacetSelectors.shadow().find('[part="value-link"]'),
-  childValueLabel: () =>
-    CategoryFacetSelectors.childValue().find('[part="value-label"]'),
-  valueCount: () =>
-    CategoryFacetSelectors.shadow().find('[part="value-count"]'),
-  parentValue: () =>
-    CategoryFacetSelectors.shadow().find('[part="parent-button"]'),
-  activeParentValue: () =>
-    CategoryFacetSelectors.shadow().find('[part="active-parent"]'),
-  activeParentValueWithText: (text: string) =>
-    CategoryFacetSelectors.shadow().find(
-      `[part="active-parent"]:contains("${text}")`
-    ),
-  showMoreButton: () =>
-    CategoryFacetSelectors.shadow().find('[part="show-more"]'),
-  showLessButton: () =>
-    CategoryFacetSelectors.shadow().find('[part="show-less"]'),
-  searchInput: () =>
-    CategoryFacetSelectors.shadow().find('[part="search-input"]'),
-  searchClearButton: () =>
-    CategoryFacetSelectors.shadow().find('[part="search-clear-button"]'),
-  searchResults: () =>
-    CategoryFacetSelectors.shadow().find('[part="search-results"]'),
-  searchResult: () =>
-    CategoryFacetSelectors.shadow().find('[part="search-result"]'),
-  searchResultPath: () =>
-    CategoryFacetSelectors.shadow().find('[part="search-result-path"]'),
-  moreMatches: () =>
-    CategoryFacetSelectors.shadow().find('[part="more-matches"]'),
-  noMatches: () => CategoryFacetSelectors.shadow().find('[part="no-matches"]'),
-  valueHighlight: () =>
-    CategoryFacetSelectors.shadow().find('[part="search-highlight"]'),
+  withId(id: string) {
+    return {
+      ...this,
+      shadow() {
+        return cy.get(`${categoryFacetComponent}[facet-id="${id}"]`).shadow();
+      },
+    };
+  },
```

Comment:
I changed every selector so they can be called either like `CategoryFacetSelectors.shadow()` or `CategoryFacetSelectors.withId('facet-a').shadow()`. This is very important to test dependsOn, since we insert multiple facets. Alternatively, I could make every selector optionally take an `id` as a parameter and pass is down to `shadow`.

Reply:
Adding the feature + the cypress tests + a cypress test refactor in the same PR makes this a pretty big review! 😄